content/tutorials/dsx-subscribe.textile

```diff
@@ -0,0 +1,249 @@
+        <th>Line</th>
+      </tr>
+    </thead>
+    <tbody>
+    </tbody>
+  </table>
+<script src="main.js"></script>
+</body>
+</html>
+```
+
+h2. Step 4 - Determining the channel names
+
+In DSX, the channels available in any product are be determined by the naming convention described below:
+
+[product:<strong>DSX-ID</strong>]:<strong>NAMESPACE</strong>:<strong>OPTIONAL-CATEGORY</strong>
+
+For the current product, we have three categories of channels, each containing a different set of data:
+
+1. Arrivals at a station, with namespace @tube:{{line}}:{{station}}:arrivals@
+2. Disruptions on a line, with namespace @tube:{{line}}:disruptions@
+3. Disruptions on all lines, with namespace @tube:disruptions@
+
+h2. Step 5 - Attaching and subscribing to a channel
+
+Let's go ahead and attach our client to the arrivals channel, i.e. the first one from the list above. If you are already a user of "Ably's DSN platform":/realtime/channels, you'll observe that the subscription is established in the same way in DSX, as in DSN.
+
+First, let's instance the Ably client as shown below. Make sure to replace the placeholder with your actual API key.
+
+```[javascript]
+var apiKey = 'YOUR-API-KEY';
+var ably = new Ably.Realtime(key: apiKey);
+```
+
+Note that in this tutorial we've used the API key directly to authenticate with Ably. This is generally "not a recommended authentication strategy":/best-practice-guide#using-api-key-client-side as the API key is prone to compromise while traveling on the internet. Ably offers "Token Authentication":/core-features/authentication#token-authentication which is more secure and is generally used in production environments. 
+
+Next, we need to add a subscription to the channel @[product:TFL/Tube]tube:' + name + ':arrivals@ where 'name' will be the name of the station chosen by the user from drop-down menu in our HTML above.
+
+In the new @main.js@ file, after declaring the required variables, let's go ahead and add a generic subscription function that'll check if the client is already attached to a channel related to one of the stations, detach if so and freshly attach to the correct channel corresponding to the currently selected station.
+
+```[javascript]
+var tableBodyTfl = document.getElementById('table-tfl').getElementsByTagName('tbody')[0];
+var channelTfl;
+
+function subscribeToChannel(name, channel, subscribeFunction) {
+  if (channel) {
+    channel.detach(function(err) {
+      if(err) {
+        console.log("Error detaching: " + err);
+      } else {
+        subscribeFunction(name);
+      }
+    });
+  } else {
+      subscribeFunction(name);
+    } 
+}
+```
+Next, let's define how we call the function above. The subscription can be set up for the default selected station initially, but we'd like our client to subscribe to a new channel whenever the user selects a new station from the drop-down menu. Add the following to your @main.js@ file below the previously added code snippet.
```

Comment:
Perhaps for the tutorial we shouldn't overcomplicate by allowing different stations to be selected, and just use one, stating the format (as you have) you can use to use any other station?